packages/parser/src/language/sankey/sankey.langium

```diff
@@ -0,0 +1,42 @@
+grammar Sankey
+
+// common stuff in ../common/common.langium
+// can't import directly because of SANKEY_LINK_SOURCE parse rule
+fragment TitleAndAccessibilities:
+  ((accDescr=ACC_DESCR | accTitle=ACC_TITLE | title=TITLE) NEWLINE+)+
+;
+
+terminal NEWLINE: /\r?\n/;
+terminal ACC_DESCR: /[\t ]*(?:accDescr(?:[\t ]*:[\t ]*[^\n\r]*?(?=%%)|\s*{[^}]*})|accDescr(?:[\t ]*:[\t ]*[^\n\r]*|\s*{[^}]*}))/;
+terminal ACC_TITLE: /[\t ]*(?:accTitle[\t ]*:[\t ]*[^\n\r]*?(?=%%)|accTitle[\t ]*:[\t ]*[^\n\r]*)/;
+terminal TITLE: /[\t ]*(?:title(?:[\t ]+[^\n\r]*?|)(?=%%)|title(?:[\t ]+[^\n\r]*|))/;
+hidden terminal WHITESPACE: /[\t ]+/;
+hidden terminal YAML: /---[\t ]*\r?\n[\s\S]*?---[\t ]*(?!.)/;
+hidden terminal DIRECTIVE: /[\t ]*%%{[\s\S]*?}%%\s*/;
+hidden terminal SINGLE_LINE_COMMENT: /[\t ]*%%[^\n\r]*(\r?\n)*/;
+
+interface Sankey {
+  accDescr?: string;
+  accTitle?: string;
+  title?: string;
+  links: SankeyLink[];
+  nodes: string[];
+}
+
+// actual grammar
+entry Sankey returns Sankey:
+  NEWLINE*
+  "sankey-beta"
+  (
+    NEWLINE* TitleAndAccessibilities links+=SankeyLink+
+    | NEWLINE+ links+=SankeyLink+
+  )
+;
+
+SankeyLink:
+  source=SANKEY_LINK_SOURCE target=SANKEY_LINK_TARGET value=SANKEY_LINK_VALUE NEWLINE+
+;
+
+terminal SANKEY_LINK_VALUE returns number: /,[\t ]*(0|[1-9][0-9]*)(\.[0-9]+)?/;
+terminal SANKEY_LINK_TARGET: /,[\t ]*("[^\n\r"]+"|[^\n\r,]+)/;
```

Comment:
* It is missing escaped quotes, if I've read it correctly. An escaped quote is `""` (double quote).
* I dont see a point of embedding spaces in grammar, we must avoid this for the sake of simplicity and ignore where possible there is not spaces in grammar.
* There is no spaces allowed between comma and the following quote

Reply:
> * It is missing escaped quotes, if I've read it correctly. An escaped quote is `""` (double quote).

It's already in the grammar, it's here:

```
"[^\n\r"]+"
```

> * I dont see a point of embedding spaces in grammar, we must avoid this for the sake of simplicity and ignore where possible there is not spaces in grammar.
> 
> * There is no spaces allowed between comma and the following quote

I added the the spaces to give the ability to parse spaces after comma and before quotes, but if you say it's not allowed then I'll remove it.

Reply:
Fixed 5a3795364db6a1fb28b759ab1e60aa3fdb3cd12f.

Reply:
How does it work actually? It must allow escaped quotes, but from what I see it allows everything except quotes. I don't see quotes supported inside quoted string.

| Sample | Output |
|--------|--------|
| "hello"  | hello |
| "hello ""world""" | hello "world" |

Reply:
I see, I haven't really seen anything like that before, not sure how to achieve that using regex.
Is there another diagram using this approach?

Reply:
Regular expressions support alternatives as well. I don't know which regexp standard langium complies with, and to what extent. But:
```
/"(""|[^"])"*/
```
or smth similar should do the trick.
Please, do not take this example as a direct guidance, it must be reworked & verified.

Reply:
Just to make sure, the double quotes could also have `%%` inside it, couldn't it?

```
sankey-beta
"source %%", target, 10
```